src/index.js

```diff
@@ -82,10 +79,63 @@ async function fetchAndParseBills() {
   // add filenames
   bills = bills.map(bill => ({
     ...bill,
-    filename: `${format(bill.date, 'YYYY-MM')}_o2.pdf`
+    filename: `${format(bill.date, 'YYYY-MM')}_o2.pdf`,
+    vendor: 'O2',
+    currency: '€',
+    metadata: {
+      importDate: new Date(),
+      version: 1
+    }
   }))
 
-  return bills
+  // now parse data in pdf files and save associated bill for each bill, to avoid save all the pdf
+  // files in memory
+  for (const bill of bills) {
+    log('info', `parsing pdf file for ${bill.date} bill`)
+    await findAndAddAmount(bill)
```

Comment:
I guess this function do a side-effect on `bill`, that is a smell. It is better if the function return a brand new updated object. Learn more about the advantages [here](https://www.sitepoint.com/functional-programming-pure-functions/):

- Reproducible Results
- Parallelization
- Memoization
- Laziness

Reply:
@enguerran Since async/await does not work in `map` I chose to do it like this, but since bluebird is already included in cozy-konnector-libs, here is another solution using it and without side effect.